zebedee/client.go

```diff
@@ -0,0 +1,302 @@
+	sdlp.FilterID = dlp.FilterID
+	bc, err := c.getBreadcrumb(dlp.URI)
+	if err != nil {
+		return StaticDatasetLandingPage{}, err
+	}
+	sdlp.Page.Breadcrumb = bc
+	sdlp.DatasetLandingPage.ParentPath = sdlp.Breadcrumb[len(sdlp.Breadcrumb)-1].Title
+
+	for index, value := range dlp.Datasets {
+		dataset := c.getDatasetDetails(value.URI)
+		dataset.IsLast = index+1 == len(dlp.Datasets)
+
+		sdlp.DatasetLandingPage.Datasets = append(sdlp.DatasetLandingPage.Datasets, dataset)
+	}
+
+	for _, value := range dlp.Alerts {
+		switch value.Type {
+		default:
+			log.Debug("Unrecognised alert type", log.Data{"alert": value})
+			fallthrough
+		case "alert":
+			sdlp.DatasetLandingPage.Notices = append(sdlp.DatasetLandingPage.Notices, datasetLandingPageStatic.Message{
+				Date:     value.Date,
+				Markdown: value.Markdown,
+			})
+		case "correction":
+			sdlp.DatasetLandingPage.Corrections = append(sdlp.DatasetLandingPage.Corrections, datasetLandingPageStatic.Message{
+				Date:     value.Date,
+				Markdown: value.Markdown,
+			})
+		}
+	}
+
+	return sdlp, nil
+}
+
+func (c ZebedeeClient) getDatasetDetails(uri string) datasetLandingPageStatic.Dataset {
+	b, err := c.get("/data?uri=" + uri)
+	if err != nil {
+		log.Error(err, nil)
+		return datasetLandingPageStatic.Dataset{URI: uri}
+	}
+
+	var d models.Dataset
+	if err = json.Unmarshal(b, &d); err != nil {
+		log.Error(err, nil)
+		return datasetLandingPageStatic.Dataset{
+			URI: uri,
+		}
+	}
+
+	var dataset datasetLandingPageStatic.Dataset
+	for _, value := range d.Downloads {
+		dataset.Downloads = append(dataset.Downloads, datasetLandingPageStatic.Download{
+			URI:       value.File,
+			Extension: strings.TrimPrefix(filepath.Ext(value.File), "."),
+			Size:      c.getFileSize(uri + "/" + value.File),
+		})
+	}
+	for _, value := range d.SupplementaryFiles {
+		dataset.SupplementaryFiles = append(dataset.SupplementaryFiles, datasetLandingPageStatic.SupplementaryFile{
+			Title:     value.Title,
+			URI:       value.File,
+			Extension: strings.TrimPrefix(filepath.Ext(value.File), "."),
+			Size:      c.getFileSize(uri + "/" + value.File),
+		})
+	}
+	dataset.Title = d.Description.Edition
+	if len(d.Versions) > 0 {
+		dataset.HasVersions = true
+	}
+
+	dataset.URI = d.URI
+
+	return dataset
+
+}
+
+func (c ZebedeeClient) getFileSize(uri string) string {
+	b, err := c.get("/filesize?uri=" + uri)
+	if err != nil {
+		log.Error(err, nil)
+		return ""
+	}
+
+	fs := struct {
+		Size int `json:"fileSize"`
+	}{}
+	if err = json.Unmarshal(b, &fs); err != nil {
+		log.Error(err, nil)
+		return ""
+	}
+
+	return datasize.ByteSize(fs.Size).HumanReadable()
+}
+
+func (c ZebedeeClient) getPageTitle(uri string) string {
+	b, err := c.get("/data?uri=" + uri + "&title")
+	if err != nil {
+		log.Error(err, nil)
+		return ""
+	}
+
+	pt := struct {
+		Title   string `json:"title"`
+		Edition string `json:"edition"`
+	}{}
+
+	if err = json.Unmarshal(b, &pt); err != nil {
+		log.Error(err, nil)
+		return ""
```

Comment:
a json unmarshal error doesn't necessarily mean we haven't got the title/edition, so possibly worth just falling through to the block below (log the error, then see what happens)